### ISSUE TITLE
Android model personalization crash with keras model

### ISSUE THREAD

**Issue by PawelFaron** [tensorflow/tensorflow#37546]
**System information** 
- Have I written custom code (as opposed to using a stock
example script provided in TensorFlow):  **- no**
- OS Platform and Distribution (e.g.,
Linux Ubuntu 16.04): Windows 10/Android
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if
the issue happens on mobile device: Android Emulator
- TensorFlow installed from (source or
binary): binary
- TensorFlow version (use command below): **2.0.0rc0** for generating model and  **org.tensorflow:tensorflow-lite:0.0.0-nightly** on the phone
- Python version: - Bazel
version (if compiling from source): Anaconda version 3.7

I'm fiddling with the example of model personalization on Android. It works great with model generated with https://github.com/tensorflow/examples/blob/master/lite/examples/model_personalization/converter/tfltransfer/tflite_transfer_convert.py but fails to work with model generated with example code given in decription of model_personnalization repo:

import tensorflow as tf
from tensorflow.keras import layers
from tensorflow.keras.regularizers import l2
from tfltransfer import bases
from tfltransfer import heads
from tfltransfer import optimizers
from tfltransfer.tflite_transfer_converter import TFLiteTransferConverter
base = bases.MobileNetV2Base(image_size=224)
head = tf.keras.Sequential([
    layers.Flatten(input_shape=(7, 7, 1280)),
    layers.Dense(
        units=32,
        activation='relu',
        kernel_regularizer=l2(0.01),
        bias_regularizer=l2(0.01)),
    layers.Dense(
        units=4,
        activation='softmax',
        kernel_regularizer=l2(0.01),
        bias_regularizer=l2(0.01)),
])
head.compile(loss='categorical_crossentropy', optimizer='sgd')
converter = TFLiteTransferConverter(4,
                                    base,
                                    heads.KerasModelHead(head),
                                    optimizers.SGD(3e-2),
                                    train_batch_size=20)
converter.convert_and_save('custom_keras_model')

The error is:
> A/libc: Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0xfffffff4 in tid 22057 (amples.transfer), pid 22057 (amples.transfer)

**Comment by Saduf2019** [comment]
@PawelFaron 
Can you please share a simple stand alone code for us to replicate the issue faced by you.

**Comment by PawelFaron** [comment]
Steps to reproduce:
1. Take android example of model personalization from https://github.com/tensorflow/examples/tree/master/lite/examples/model_personalization/android
2. Change the model in assets to the one created by code provided in model personalization https://github.com/tensorflow/examples/tree/master/lite/examples/model_personalization#converting-the-custom-keras-model-using-the-python-api
3. This will trigger the crash.

**Comment by Saduf2019** [comment]
@PawelFaron 
we ran the code and face a different error, please refer to [this gist](https://colab.sandbox.google.com/gist/amahendrakar/46ff1ade36232f44cda6fa5fda9e08e7/untitled81.ipynb#scrollTo=16vjeC2sMegq) for the same

**Comment by PawelFaron** [comment]
You have the tfltransfer in the repo https://github.com/tensorflow/examples/tree/master/lite/examples/model_personalization/converter/tfltransfer

**Comment by jaeyoo** [comment]
@Saduf2019  To run the example, `python setup.py install` is prerequisite, then your python can recognize tfltransfer.
@PawelFaron, I've tried to run it on my side, but it worked. You mention that you used Android Emulator. Could you specify more details about AVD setup? I could run it on both my Pixel 3a XL and its emulator,

**Comment by Saduf2019** [comment]
@@PawelFaron,
could you please update as per above comment
